pkg/network/events/monitor_windows.go

```diff
@@ -0,0 +1,79 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package events handles process events
+package events
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	"go4.org/intern"
+
+	"github.com/DataDog/datadog-agent/pkg/security/secl/model"
+	"github.com/DataDog/datadog-agent/pkg/util/winutil/iisconfig"
+)
+
+func getProcessStartTime(ev *model.Event) time.Time {
+	if ev.GetEventType() == model.ExecEventType {
+		return ev.GetProcessExecTime()
+	}
+	// windows does not have a fork event.
+	return time.Time{}
+}
+
+func makeTagsSlice(apmtags iisconfig.APMTags) []*intern.Value {
+	tags := make([]*intern.Value, 0, 3)
+	if len(apmtags.DDService) > 0 {
+		tags = append(tags, intern.GetByString("service"+":"+apmtags.DDService))
+	}
+	if len(apmtags.DDEnv) > 0 {
+		tags = append(tags, intern.GetByString("env"+":"+apmtags.DDEnv))
+	}
+	if len(apmtags.DDVersion) > 0 {
+		tags = append(tags, intern.GetByString("version"+":"+apmtags.DDVersion))
+	}
+	if len(tags) == 0 {
+		return nil
+	}
+	return tags
+}
+func getAPMTags(filename string) []*intern.Value {
+
+	dir := filepath.Dir(filename)
+	if dir == "" {
+		return nil
+	}
+
+	// see if there's an app.config in the directory
+	appConfig := filepath.Join(dir, "app.config")
+	ddJSON := filepath.Join(dir, "datadog.json")
+	if _, err := os.Stat(appConfig); err == nil {
```

Comment:
Should the error be log somewhere?
In case the file is missing, restricted, corrupted, locked, etc?

Reply:
The file is expected to be missing; in fact that's going to be the case in the vast majority of the times this code is executed.

This code path is also expected to be executed a _lot_ (on every process start), so i felt it was better to just continue processing in the case where the file exists (stat returns without error) rather than trying to decode specific errors with a lot of extra logic.

Reply:
Can we ignore only the case when the file is missing and not other errors?
How do engineers troubleshoot file related problems in general?

Reply:
Offline discussion, this is a hot path and maybe too noisy.